patches/api/0391-Fix-wrong-position-for-bed-spawn-loc.patch

```diff
@@ -0,0 +1,92 @@
+      *
+-     * @return Bed Spawn Location if bed exists, otherwise null.
++     * @return Block Spawn Location if the block exists, otherwise null.
+      */
+     @Nullable
+-    public Location getBedSpawnLocation();
+-    // Paper start
++    default Location getBedSpawnLocation() {
++        return getBedSpawnLocation(false); // Keep old behavior for offline
++    }
++
++    /**
++     * Gets the Location where the player will spawn at their spawn block (bed, respawn anchor...), null if
++     * they have not slept in one.
++     *
++     * @param load Load the current location to retrieve the exact position of the spawn block
++     *             and check if this position is valid or not
++     * @return Block Spawn Location if the block exists, otherwise null.
++     */
++    @Nullable
++    Location getBedSpawnLocation(boolean load);
++
+     /**
+      * Gets the last date and time that this player logged into the server.
+      * <p>
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index b607c229cfb1e95b17b6a0073380089ef5e1b675..c43617e43995d8fec0a19cfbdb5e0fe8af397677 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -351,31 +351,35 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     public boolean isSleepingIgnored();
+ 
++    // Paper start
+     /**
+-     * Gets the Location where the player will spawn at their bed, null if
+-     * they have not slept in one or their current bed spawn is invalid.
++     * Gets the Location where the player will spawn at their spawn block (bed, respawn anchor...),
++     * null if they have not slept in one or their current spawn block is invalid.
+      *
+-     * @return Bed Spawn Location if bed exists, otherwise null.
++     * @return Block Spawn Location if the block exists, otherwise null.
+      */
+     @Nullable
+     @Override
+-    public Location getBedSpawnLocation();
++    default Location getBedSpawnLocation() {
++        return getBedSpawnLocation(true);
++    }
+ 
+     /**
+-     * Sets the Location where the player will spawn at their bed.
++     * Sets the Location where the player will spawn at their spawn block (bed, respawn anchor...).
+      *
+      * @param location where to set the respawn location
+      */
+     public void setBedSpawnLocation(@Nullable Location location);
+ 
+     /**
+-     * Sets the Location where the player will spawn at their bed.
++     * Sets the Location where the player will spawn at their spawn block (bed, respawn anchor...).
+      *
+      * @param location where to set the respawn location
+      * @param force whether to forcefully set the respawn location even if a
+-     *     valid bed is not present
++     *     valid spawn block is not present
+      */
+     public void setBedSpawnLocation(@Nullable Location location, boolean force);
++    // Paper end
+ 
```

Comment:
This is a huge paper block for single line changes in javadocs.
Probably best to add in line comments. Html comments work fine in javadocs for that matter. (See https://github.com/PaperMC/Paper/blob/master/patches/api/0226-Clarify-the-Javadocs-for-Entity.getEntitySpawnReason.patch#L16)

Reply:
Ooh, i didn't know that was allowed